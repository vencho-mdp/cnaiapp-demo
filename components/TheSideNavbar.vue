<template>
  <div class="min-h-screen md:flex" data-dev-hint="container">
    <aside
      id="sidebar"
      class="bg-primary-darkblue flex w-full px-8 gap-16 items-center overflow-y-auto justify-around md:flex-col md:px-0 py-3 md:pt-10 md:pb-96 md:w-16 md:!justify-start"
    >
      <span v-for="link in filtered_links" :key="link.name">
        <nuxt-link :to="`/panel/${link.route}`">
          <img
            class="hover-effect"
            :src="getImgUrl(link.img)"
            :alt="link.route"
          />
        </nuxt-link>
      </span>
      <a class="cursor-pointer hover-effect" @click="handleLogout()">
        <img :src="getImgUrl('log-out')" alt="Cerrar sesiÃ³n" />
      </a>
    </aside>

    <main id="content" class="flex-1 max-h-full">
      <slot />
    </main>
  </div>
</template>

<script>
export default {
  name: "SideNavbar",
  data() {
    return {
      links: [
        {
          img: "home",
          route: "",
          should_render: true,
        },
        {
          img: "take-attendance",
          route: "asistencia-alumnos",
          should_render:
            (this.$store.state.authentication.user_data?.groups.includes(
              "preceptor"
            ) &&
              this.$store.state.authentication.user_data.classes_ids.length >
                0) ||
            this.$store.state.authentication.user_data?.groups.includes(
              "management_team"
            ) ||
            this.$store.state.authentication.user_data?.subjects?.some(
              (subject) =>
                this.$store.state.EXTRA_CURRICULAR_SUBJECTS.includes(subject)
            ),
        },
        // {
        //   img: "users",
        //   route: "usuarios",
        //   should_render:
        //     this.$store.state.authentication.user_data.groups.includes(
        //       "management_team"
        //     ),
        // },
        {
          img: "auditory",
          route: "auditoria",
          should_render:
            this.$store.state.authentication.user_data.groups.includes(
              "management_team"
            ),
        },
      ],
    };
  },
  computed: {
    filtered_links() {
      return this.links.filter((link) => link.should_render);
    },
  },
  methods: {
    getImgUrl(name) {
      const images = require.context("../assets/images", false, /\.svg$/);
      return images("./" + name + ".svg");
    },
    handleLogout() {
      this.$router.replace({
        path: "/iniciar-sesion",
      });
      setTimeout(() => {
        this.$store.dispatch("authentication/logout");
      }, 100);
    },
  },
};
</script>

<style scoped></style>
