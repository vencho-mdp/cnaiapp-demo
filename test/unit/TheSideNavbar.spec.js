import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import TheSideNavbar from '@/components/TheSideNavbar.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('TheSideNavbar', () => {
  // eslint-disable-next-line no-unused-vars
  let actions
  let state
  let store

  beforeEach(() => {
    actions = {
      'authentication/logout': jest.fn()
    }

    state = {
      authentication: {
        user_data: {
          classes_ids: [1, 2, 3],
          groups: ['preceptor']
        }
      }
    }

    store = new Vuex.Store({
      actions,
      state
    })
  })

  it('renders the correct markup', () => {
    const component = shallowMount(TheSideNavbar, {
      store,
      localVue
    })
    expect(component.html).toMatchSnapshot()
  })

  it('logs out', async () => {
    const component = shallowMount(TheSideNavbar, {
      store,
      localVue,
      mocks: {
        $router: {
          replace: jest.fn()
        }
      }
    })
    await component.find('aside > a').trigger('click')
    expect(actions['authentication/logout']).toHaveBeenCalled()
  })

  it('has links', () => {
    const component = shallowMount(TheSideNavbar, {
      store,
      localVue
    })
    expect(component.findAll('a').length).toBeGreaterThan(0)
  })
})
