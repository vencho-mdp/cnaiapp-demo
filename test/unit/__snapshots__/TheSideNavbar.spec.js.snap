// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSideNavbar renders the correct markup 1`] = `[Function]`;
